independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BuiltinBean.java

```diff
@@ -311,10 +311,16 @@ private static void generateBeanManagerBytecode(GeneratorContext ctx) {
     private static void generateResourceBytecode(GeneratorContext ctx) {
         ResultHandle annotations = ctx.constructor.newInstance(MethodDescriptor.ofConstructor(HashSet.class));
         // For a resource field the required qualifiers contain all annotations declared on the field
+        // (hence we need to check if they are runtime-retained and their classes are available)
         if (!ctx.injectionPoint.getRequiredQualifiers().isEmpty()) {
             for (AnnotationInstance annotation : ctx.injectionPoint.getRequiredQualifiers()) {
-                // Create annotation literal first
+                if (!annotation.runtimeVisible()) {
```

Comment:
Can this actually happen for a built-in bean? I would assume in those cases the annotation is always present and is runtime visible. What am I missing?

Reply:
You're missing the comment a few lines above :-) This code is also used for resource injection, when all annotations on the injection point are considered "qualifiers".